### PR TITLE
feat: tx sign --sign-with-lab

### LINKED ISSUE #1428: Enable the CLI to generate links or files compatible with the Stellar Lab
## What problem does your feature solve?
To transactions created in the CLI, allow users to connect to a wallet not stored by the CLI. This is being explicitly asked by developer

## What would you like to see?
- Use Wallet kit (https://github.com/Creit-Tech/Stellar-Wallets-Kit) - Wallet kit is already implemented in Lab
- Land user in [Lab](https://beta-laboratory.stellar.org/?$=network$id=testnet&label=Testnet&horizonUrl=https:////horizon-testnet.stellar.org&rpcUrl=https:////soroban-testnet.stellar.org&passphrase=Test%20SDF%20Network%20/;%20September%202015;;) with wallet kit open 
- Note: Lab will use https://lab.stellar.org as the URL when launched

This issue includes pass the xdr to the sign page on Lab and pass back the signed xdr to the CLI.

### PR DESCRIPTION
**Pull request by elizabethengelman**
This branch is based on https://github.com/stellar/stellar-cli/pull/1590.

Companion Lab PR: https://github.com/stellar/laboratory/pull/987 (merged)

old pr: https://github.com/stellar/stellar-cli/pull/1579

### What
Closes #1428 

This PR adds the ability to sign a tx via lab, by opening lab in the browser with the xdr prefilled in the Lab UI.

To test, on this branch, run:
- you can generate your tx xdr, but this example uses xdr created from deploying the test_hello_world contract.
(`stellar contract deploy --wasm target/wasm32-unknown-unknown/test-wasms/test_hello_world.wasm --build-only --network testnet --source testnet-alice`)
- run the following from the root dir of this project, on this branch:
```
echo AAAAAgAAAADcOHnq5sGLOngOCEMyLqqn5CvFV2HGbOSjJAIzhqBdkAAAAGQAEb7UAAAAAQAAAAAAAAAAAAAAAQAAAAAAAAAYAAAAAQAAAAAAAAAAAAAAANw4eermwYs6eA4IQzIuqqfkK8VXYcZs5KMkAjOGoF2Q04SY/PMseMCgfbEFkX5gTr9qD/d0mvD9H3acFOz/X5oAAAAALtGgcgsAibk5q8VOLY2R1G1NdGX7fLzlz7iBFBy64JIAAAAAAAAAAAAAAAA= \
| cargo run tx sign --network testnet \
--sign-with-lab \
--lab-url http://laboratory-pr987.previews.kube001.services.stellar-ops.com/transaction/sign
```

### Why

This will allow for another way for users to sign their tx. 

### Known limitations

* Based on discussions during our weekly check-in, this PR is the first step in implementing this feature. For now, when the user adds the `--sign-with-lab` flag, the CLI will send the transaction's xdr to lab via a query param, and then will print out the xdr to the stdout. 

    In future iterations, we may want to send the signed tx xdr back to the command line, but for now the intention is to have the user either:
    a. send the transaction from lab
    b. manually copy the signed xdr from Lab, and paste it back in the command line to use with the cli
* Instead of having the flag be `--sign-with-lab`, what if we change it to `--sign-externally`? This would open us up to allow a user to pass in any URL that accepts `xdr` and `networkPasshrase` as query params. 🤔

### CODE REVIEW

File: cmd/soroban-cli/src/config/sign_with.rs [hunk 1]
```diff
@@ -101,4 +117,48 @@ impl Args {
             .account(&self.source_account)?
             .public_key(self.hd_path)?)
     }
+
+    pub fn sign_tx_env_with_lab(
+        &self,
+        network: &Network,
+        tx_env: &TransactionEnvelope,
+    ) -> Result<(), Error> {
+        if !self.yes {
```
Comment (elizabethengelman) [inline cmd/soroban-cli/src/config/sign_with.rs:126]: Just wanted to flag this - do we still want to do this for `sign-with-lab`? I think that we would run into a similar issue that @willemneal fixed in https://github.com/stellar/stellar-cli/pull/1590 with the stdin reader. 

More context: https://github.com/AhaLabs/stellar-cli/pull/25#discussion_r1755504162

File: cmd/soroban-cli/src/config/sign_with.rs [hunk 2]
```diff
@@ -37,14 +46,17 @@ pub struct Args {
...
+    #[arg(long, conflicts_with = "sign_with_key", env = "STELLAR_SIGN_WITH_LAB")]
+    pub sign_with_lab: bool,
+    /// Lab URL for `sign_with_lab`
     #[arg(
         long,
         conflicts_with = "sign_with_key",
-        env = "STELLAR_SIGN_WITH_LAB",
-        hide = true
+        env = "STELLAR_SIGN_WITH_LAB_URL",
+        default_value = "https://lab.stellar.org/transaction/cli-sign"
```
Comment (leighmcculloch) [inline cmd/soroban-cli/src/config/sign_with.rs:57]: I suggest as a starting point we don't make this configurable and make it hardcoded. There's only one lab deployment, we don't know of anyone else running their own lab, and quickstart doesn't yet embed lab. It's easy to add this parameter in down the track once quickstart is hosting lab or if someone adds it.

**Review by leighmcculloch — COMMENTED** [review]
@elizabethengelman could you merge in the tx sign branch to get the latest here and then ping me for a review?

File: cmd/soroban-cli/src/config/sign_with.rs (hunk 2 above)
Comment (willemneal) [reply cmd/soroban-cli/src/config/sign_with.rs:51]: Good point, I'll remove it now

File: cmd/soroban-cli/src/config/sign_with.rs (hunk 1 above)
Comment (willemneal) [reply cmd/soroban-cli/src/config/sign_with.rs]: It's removed!

**Review by leighmcculloch — COMMENTED** [review]
I merged main into this PR to update it.

Some comments below.

**Review by leighmcculloch — APPROVED** [review]

**Review by leighmcculloch — APPROVED** [review]

File: cmd/soroban-cli/src/signer.rs [hunk 3]
```diff
@@ -251,3 +259,28 @@ impl LocalKey {
...
+        let mut url = url::Url::parse(Self::URL)?;
+        url.query_pairs_mut()
+            .append_pair("networkPassphrase", &network.network_passphrase)
+            .append_pair("xdr", &xdr);
+        let url = url.to_string();
+
+        printer.globeln(format!("Opening lab to sign transaction: {url}"));
+        open::that(url)?;
+
+        Err(Error::ReturningSignatureFromLab)
```
Comment (Ifropc) [inline cmd/soroban-cli/src/signer.rs:284]: In the future it'd be good to add a TODO with a task number in the code

**Pull request merged**